### PR TITLE
Init code style checking, pytest, and coverage

### PR DESCRIPTION
**Pull request by dachengx**
Also, add a bug report template.

The bot still complains about the docstring of functions and classes. The beautifying of the docstring will be in another separate PR. It also complains about `assert`, we should rethink about whether raise errors instead, but also in another PR.

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install pytest coverage coveralls
+          pip install -r requirements.txt
+
+      - name: Install Appletree
```
Comment (kdund) [inline .github/workflows/pytest.yml:49]: is this an oversight?

Comment (kdund) [reply .github/workflows/pytest.yml:49]: perhaps just a copy-pasted comment :)

**Review by kdund — APPROVED** [review]
Looks good to me -- need to rename appletree to alea, and then good :)?

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (dachengx) [reply .github/workflows/pytest.yml]: Just fixed.

**Pull request merged**